### ISSUE TITLE
1.0.0.alpha2 errors compound

### ISSUE THREAD

**Issue by chevinbrown** [dry-rb/dry-validation#477]
Calling same variable assigned to `Dry::Validation::Contract` compounds error messages. 

```
class MyKlass < Dry::Validation::Contract
  params do
    required(:id) { int? }
    required(:duration) { int? }
  end
```

`result = MyKlass.new.call(id: nil)`
Each time `result` is used, error messages are still added. 

I haven't had time to jump into the 1.0.0.alpha codebase, but it seems something should `throw` or be memoized? 

Just wanted to get this report listed, but happy to help diagnose further later next week!

**Comment by solnic** [comment]
``` ruby
require 'dry/validation'

class MyKlass < Dry::Validation::Contract
  params do
    required(:id) { int? }
    required(:duration) { int? }
  end
end

c = MyKlass.new

puts c.(id: nil).inspect 
puts c.(id: nil).inspect
puts c.(id: nil).inspect

r = c.(id: nil)

puts r.errors.inspect
puts r.errors.inspect
puts r.errors.inspect
```

output:

``` ruby
#<Dry::Validation::Result{:id=>nil} errors={:id=>["must be an integer"], :duration=>["is missing"]}>
#<Dry::Validation::Result{:id=>nil} errors={:id=>["must be an integer"], :duration=>["is missing"]}>
#<Dry::Validation::Result{:id=>nil} errors={:id=>["must be an integer"], :duration=>["is missing"]}>
#<Dry::Validation::ErrorSet messages=[#<Dry::Schema::Message predicate=:int? path=[:id] text="must be an integer" options={:args=>[], :input=>nil, :rule=>nil}>, #<Dry::Schema::Message predicate=:key? path=[:duration] text="is missing" options={:args=>[:duration], :input=>#<Dry::Schema::Result{:id=>nil} errors={:id=>["must be an integer"], :duration=>["is missing"]}>, :rule=>:duration}>] options={:failures=>true}>
#<Dry::Validation::ErrorSet messages=[#<Dry::Schema::Message predicate=:int? path=[:id] text="must be an integer" options={:args=>[], :input=>nil, :rule=>nil}>, #<Dry::Schema::Message predicate=:key? path=[:duration] text="is missing" options={:args=>[:duration], :input=>#<Dry::Schema::Result{:id=>nil} errors={:id=>["must be an integer"], :duration=>["is missing"]}>, :rule=>:duration}>] options={:failures=>true}>
#<Dry::Validation::ErrorSet messages=[#<Dry::Schema::Message predicate=:int? path=[:id] text="must be an integer" options={:args=>[], :input=>nil, :rule=>nil}>, #<Dry::Schema::Message predicate=:key? path=[:duration] text="is missing" options={:args=>[:duration], :input=>#<Dry::Schema::Result{:id=>nil} errors={:id=>["must be an integer"], :duration=>["is missing"]}>, :rule=>:duration}>] options={:failures=>true}>
```

seems like it's fixed in master?

(`errors` now returns `ErrorSet`)

**Comment by chevinbrown** [comment]
You're example is correct, but this example will compound errors: 
```
r = c.(id: nil)

puts r
puts r

#<Dry::Validation::Result{} errors={:id=>["is missing"], :duration=>["is missing"]}>
#<Dry::Validation::Result{} errors={:id=>["is missingis missing"], :duration=>["is missingis missing"]}>
```

**Comment by solnic** [comment]
OK thank you. I'll fix it in the next alpha.
